In [28]:
import pandas as pd
import sys

sys.path.append("../src")

from data_loader import download_prices_adj_close
from risk_metrics import compute_returns, compute_volatility, compute_sharpe_ratio

# Récupération des prix ajustés
#Attention on ne peut faire qu'un an de periode avec la fonction que l'on a créer
tickers = ["AAPL", "MSFT", "GOOGL"]
prices = download_prices_adj_close(tickers, start="2024-08-01", end="2025-07-31")

# Calculs
returns = compute_returns(prices)
volatility = compute_volatility(returns)
sharpe = compute_sharpe_ratio(returns)

# Affichage des résultats

print("\nSharpe Ratio annualisé :\n", sharpe)
print("\nNombre de jours observés :", len(returns))


[*********************100%***********************]  3 of 3 completed


Sharpe Ratio annualisé :
 Ticker
AAPL    -0.124461
GOOGL    0.466199
MSFT     0.878148
dtype: float64

Nombre de jours observés : 248


#### Sharpe calculé pour Apple (AAPL) : -0.12  
#### Sharpe selon PortfoliosLab : –0,13

#### Source d'erreur : Nombre de jours observés = 248 != 252 
